ArduPlane/mode_guided.cpp

```diff
@@ -139,13 +139,17 @@ void ModeGuided::update_target_altitude()
         // then scale x100 to match last_target_alt and convert to a signed int32_t as it may be negative
         int32_t delta_amt_i = (int32_t)(100.0 * delta_amt_f); 
         Location temp {};
-        temp.alt = plane.guided_state.last_target_alt + delta_amt_i; // ...to avoid floats here, 
+        int32_t temp_alt = plane.guided_state.last_target_alt + delta_amt_i;
```

Comment:
```suggestion
        int32_t temp_alt = constrain_int32(plane.guided_state.target_alt, plane.guided_state.last_target_alt - delta_amt_i,  plane.guided_state.last_target_alt + delta_amt_i);
```
You can use the constrain trick so we don't need to know the sign of the accel.

Reply:
That removes the if and the else.

Reply:
Infact, this whole thing is velocity limiting not accel limiting. The command calls for a accel limit. There is either a error on the MAVLink spec or in the code.

Reply:
Would it not be better to compare the new target location to the **current location** rather than the last target to determine the velocity required

Reply:
Thanks -done.